umbraco/Umbraco-CMS
# V8: Don't allow logins for users with no content and/or media start nodes

### Prerequisites


If there's an existing issue for this PR then this fixes https://github.com/umbraco/Umbraco-CMS/issues/5641

### Description

This PR ensures that users that belong to groups that do not define any content and/or media start nodes are not allowed to log in to the back office.

If a login for such a user is attempted, the login will fail and a message will be added to the log along the lines of: `Login attempt failed for username [username] from IP address [user IP], no content and/or media start nodes could be found for any of the user's groups`.

![image](https://user-images.githubusercontent.com/7405322/61317058-836efa00-a802-11e9-8785-7610ddc55493.png)

/cc @Shazwazza 

#### Testing this PR

1. Create a new group that has no content and media start nodes configured.
2. Create a new user and make it a member of that group (and no other groups).
3. Verify that the user can't log in to the back office.
4. Assign any other group to the user as well (that is, any group _with_ content and media start nodes configured).
5. Verify that the user can log in.
